src/node.rs

```diff
@@ -112,10 +129,42 @@ impl Node {
 
         Some((class_name, class_version))
     }
+
+    /*
+    pub fn get_basic_as_str(&self) -> &'static str {
+        match NodeBasic::from_u8(self.get_basic()) {
+            Some(NodeBasic::Controller) => "Controller",
+            Some(NodeBasic::StaticController) => "Static Controller",
+            Some(NodeBasic::Slave) => "Slave",
+            Some(NodeBasic::RoutingSlave) => "Routing Slave",
+            _ => "unknown"
+        }
+    }
+    */
+}
+
+#[derive(Debug)]
+struct Neighbor {
+    home_id: u32,
+    node_id: u8
+}
+
+impl Neighbor {
+    pub fn from_node(node: &Node) -> Neighbor {
+        Neighbor { home_id: node.get_home_id(), node_id: node.get_id() }
+    }
 }
```

Comment:
not really convinced that we need a dedicated struct for this... this is just a Node and you created this struct only to be able to easily format a debug string. But I think you can just add a `simple_debug` method to `Node` that would output `format!("Node {{ home_id: {}, node_id {} }}", self.home_id, self.node_id)` and call it.